lib/src/backend.rs

```diff
@@ -143,7 +144,14 @@ content_hash! {
     pub struct Commit {
         pub parents: Vec<CommitId>,
         pub predecessors: Vec<CommitId>,
-        pub root_tree: TreeId,
+        pub root_tree: conflicts::Conflict<TreeId>,
+        /// Indicates that there this commit uses the new tree-level conflict format, which means
+        /// that if `root_tree` is not a conflict, we know that we won't have to walk it to
+        /// determine if there are conflicts.
+        // TODO(#1624): Delete this field at some point in the future, when we decide to drop
+        // support for conflicts in older repos, or maybe after we have provided an upgrade
+        // mechanism.
+        pub tree_conflict_format: bool,
```

Comment:
Optional: I find this name a bit confusing. For the proto, I haven't had much better ideas that `tree_level_conflicts` since it should default to `false` (but see another comment). Here OTOH perhaps a `legacy_path_level_conflict` would be a bit better.

Reply:
I renamed this field to `uses_tree_conflict_format` to match the proto. I hope that works for you.